sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -400,74 +456,75 @@ public synchronized void onActivityStarted(final @NotNull Activity activity) {
     }
   }
 
+  @Override
+  public void onActivityPostStarted(final @NotNull Activity activity) {
+    final @Nullable ActivityLifecycleTimeSpan timeSpan = activityLifecycleMap.remove(activity);
+    if (appStartSpan == null) {
+      return;
+    }
+    if (timeSpan != null) {
+      timeSpan.getOnStart().stop();
+      timeSpan.getOnStart().setDescription(activity.getClass().getName() + ".onStart");
+      AppStartMetrics.getInstance().addActivityLifecycleTimeSpans(timeSpan);
+    }
+  }
+
   @Override
   public synchronized void onActivityResumed(final @NotNull Activity activity) {
+    if (!isAllActivityCallbacksAvailable) {
+      onActivityPostStarted(activity);
+    }
     if (performanceEnabled) {
-
       final @Nullable ISpan ttidSpan = ttidSpanMap.get(activity);
       final @Nullable ISpan ttfdSpan = ttfdSpanMap.get(activity);
-      final View rootView = activity.findViewById(android.R.id.content);
-      if (rootView != null) {
+      if (activity.getWindow() != null) {
```

Comment:
Do we still need this check? Since we have a reference to the activity, we could always call `irstDrawDoneListener.registerForNextDraw`

Reply:
It's a safety guard: if the window is null (i'm not sure it ever happens), the `FirstDrawDoneListener.registerForNextDraw` returns early, and the ttid is never stopped.
In this case i just post the ttid finish on the main thread, as it should be good enough